kuma/users/views.py

```diff
@@ -138,27 +147,192 @@ def ban_user_and_cleanup_summary(request, user_id):
+
+    actions_taken = {
+        'revisions_reported_as_spam': submitted_to_akismet_by_distinct_doc,
+        'revisions_reverted_list': revisions_reverted_by_distinct_doc,
+        'revisions_deleted_list': revisions_deleted_by_distinct_doc
+    }
+
+    """ The "Needs followup" section """
+    # TODO: Phase IV: If user made actions while reviewer was banning them
+    new_action_by_user = []
+    needs_follow_up = {
+        'manual_revert': new_action_by_user,
+        'not_submitted_to_akismet': not_submitted_to_akismet_by_distinct_doc,
+        'not_reverted_list': revisions_not_reverted_by_distinct_doc,
+        'not_deleted_list': revisions_not_deleted_by_distinct_doc
+
+    }
+
+    """ The "No Actions Taken" section """
+    revisions_already_spam = revisions_from_last_three_days.filter(
+        id__in=request.POST.getlist('revision-already-spam')
+    )
+    revisions_already_spam = [rev for rev in revisions_already_spam]
+    revisions_already_spam_by_distinct_doc = revision_by_distinct_doc(revisions_already_spam)
+
+    revisions_already_spam_ids = [rev.id for rev in revisions_already_spam]
+
+    not_identified_as_spam = revisions_from_last_three_days.exclude(
+        id__in=revisions_already_spam_ids).exclude(
+        id__in=revisions_to_mark_as_spam_and_revert.values_list('id', flat=True))
+    not_identified_as_spam = [rev for rev in not_identified_as_spam]
+    not_identified_as_spam_by_distinct_doc = revision_by_distinct_doc(not_identified_as_spam)
+
+    no_actions_taken = {
+        'revisions_already_identified_as_spam': revisions_already_spam_by_distinct_doc,
+        'revisions_not_identified_as_spam': not_identified_as_spam_by_distinct_doc
+    }
 
     context = {'detail_user': user,
                'form': UserBanForm(),
-               'revisions_reverted': revisions_reverted,
-               'revisions_needing_follow_up': revisions_needing_follow_up}
+               'actions_taken': actions_taken,
+               'needs_follow_up': needs_follow_up,
+               'no_actions_taken': no_actions_taken
+               }
 
     return render(request,
                   'users/ban_user_and_cleanup_summary.html',
                   context)
 
 
+def revision_by_distinct_doc(list_of_revisions):
+    if len(list_of_revisions):
+        list_of_revisions_distinct_doc = list(
+            Document.admin_objects.filter(
+                revisions__in=list_of_revisions
+            ).annotate(
+                latest_rev=Max('revisions')
+            ).values_list('latest_rev', flat=True)
+        )
+        list_of_revisions_distinct_doc = [rev for rev in list_of_revisions if rev.id in list_of_revisions_distinct_doc]
+    else:
+        list_of_revisions_distinct_doc = []
+
+    return list_of_revisions_distinct_doc
+
+
+@permission_required('users.add_userban')
+def revert_document(request, document_path, revision_id):
+    """
+    Revert document to a specific revision.
```

Comment:
I see that this is the logic from ``kuma/wiki/views/delete.py``, modified for this context. The idea is sound, but I think the interface is wrong. You already have the ``revision`` and the ``document``, but by using ``locale_and_slug_from_path`` you bring in a lot of logic that could pick a different document based on the spam moderator's preferred locale.  Instead, pass in the ``revision`` and go from there. When you are done, there may or may not be enough to justify a second function.